File: app/src/main/java/ru/SnowVolf/convertx/ui/activity/BaseActivity.java
package ru.SnowVolf.convertx.ui.activity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import ru.SnowVolf.convertx.settings.SettingsActivity;
import ru.SnowVolf.convertx.ui.Interfacer;

public class BaseActivity extends AppCompatActivity {
    private static BaseActivity INSTANCE = null;
    SharedPreferences preferences;
    //Theme
    private final BroadcastReceiver mThemeReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            if (SettingsActivity.class.equals(BaseActivity.this.getClass())){
                finish();
                startActivity(getIntent());
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
            } else recreate();
        }
    };
    public BaseActivity(){

    }

    public static BaseActivity getInstance() {
        if (INSTANCE == null) {
            new BaseActivity();
        }
        return INSTANCE;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        preferences = PreferenceManager.getDefaultSharedPreferences(this);
        LocalBroadcastManager.getInstance(this).registerReceiver(mThemeReceiver, new IntentFilter("org.openintents.action.REFRESH_THEME"));
        Interfacer.applyTheme(this);
        super.onCreate(savedInstanceState);
        INSTANCE = this;
    }

    @Override
    protected void onDestroy() {
        LocalBroadcastManager.getInstance(this).unregisterReceiver(mThemeReceiver);
        super.onDestroy();
    }

}

File: app/src/main/java/ru/SnowVolf/convertx/ui/activity/BaseActivity.kt
package ru.SnowVolf.convertx.ui.activity

import android.content.*
import androidx.appcompat.app.AppCompatActivity
import ru.SnowVolf.convertx.settings.SettingsActivity
import android.os.Bundle
import android.preference.PreferenceManager
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import ru.SnowVolf.convertx.ui.Interfacer

open class BaseActivity : AppCompatActivity() {
    var preferences: SharedPreferences? = null

    //Theme
    private val mThemeReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            if (SettingsActivity::class.java == this@BaseActivity.javaClass) {
                finish()
                startActivity(getIntent())
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
            } else recreate()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        preferences = PreferenceManager.getDefaultSharedPreferences(this)
        LocalBroadcastManager.getInstance(this).registerReceiver(mThemeReceiver, IntentFilter("org.openintents.action.REFRESH_THEME"))
        Interfacer.applyTheme(this)
        super.onCreate(savedInstanceState)
        INSTANCE = this
    }

    override fun onDestroy() {
        LocalBroadcastManager.getInstance(this).unregisterReceiver(mThemeReceiver)
        super.onDestroy()
    }

    companion object {
        private var INSTANCE: BaseActivity? = null
        val instance: BaseActivity?
            get() {
                if (INSTANCE == null) {
                    BaseActivity()
                }
                return INSTANCE
            }
    }
}